platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/InputField.kt

```diff
@@ -71,6 +74,20 @@ internal fun InputField(
             when (interaction) {
                 is FocusInteraction.Focus -> inputFieldState = inputFieldState.copy(focused = true)
                 is FocusInteraction.Unfocus -> inputFieldState = inputFieldState.copy(focused = false)
+                is PressInteraction.Press -> inputFieldState = inputFieldState.copy(pressed = true)
+                is PressInteraction.Cancel,
+                is PressInteraction.Release -> inputFieldState = inputFieldState.copy(pressed = false)
+            }
+        }
+    }
+
+    // Note: we need a separate interaction source due to b/466012773 (Jetpack Compose issue)
+    val hoverInteractionSource = remember { MutableInteractionSource() }
+    LaunchedEffect(hoverInteractionSource) {
+        hoverInteractionSource.interactions.collect { interaction ->
+            when (interaction) {
+                is HoverInteraction.Enter -> inputFieldState = inputFieldState.copy(hovered = true)
+                is HoverInteraction.Exit -> inputFieldState = inputFieldState.copy(hovered = false)
```

Comment:
Not something we need to change in this PR, but I've also saw other pattern to reduce duplications a bit. Maybe we can streamline them:

```
is HoverInteraction ->
    inputFieldState = inputFieldState.copy(hovered = interaction is HoverInteraction.Enter)
```